lib/stdlib/src/binary.erl

```diff
@@ -934,6 +935,28 @@ unhex(X) ->
              no, 10, 11, 12, 13, 14, 15, no, no, no, no, no, no, no, no, no  %96
             }).
 
+-doc """
+Joins a list of binaries together by a specified Separator.
+Equivalent to `iolist_to_binary(lists:join(Separator, Binaries))` with binary:join/2 being the faster of the two.
+
+_Example:_
+
+```erlang
+1> binary:join([<<"a">>, <<"b">>, <<"c">>], <<", ">>).
+<<"a, b, c">>
+```
+""".
+-doc(#{since => <<"OTP 27.0">>}).
+-spec join([binary()], binary()) -> binary().
+join([H], _Separator) -> H;
+join([H | T], Separator) ->
+    join(T, Separator, H).
```

Comment:
as an option
```suggestion
    lists:foldl(fun(Element, Acc) -> <<Acc/binary, Separator/binary, Element/binary>> end, H, T).
```

Reply:
Would that not lose some performance though? Performance being the main reason for this implementation.